mod/cornercutter/SpawnCollection.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace cornercutter
+{
+    class SpawnCollection
+    {
+        public List<WeightedSkill> Skills { get; private set; }
+        // Boolean for whether we should wrap around when pulling this pool or treat it as a bucket to grab from
+        private readonly bool isLooping;
+        // For now, where isLooping is set, currentIndex is used, otherwise currentTotal for weight tracking
+        // Seperated for clarity
+        private int currentIndex;
+        private int currentTotal;
+        private readonly Random rng;
+
+        public SpawnCollection(bool loopSpawns)
+        {
+            Skills = new List<WeightedSkill>();
+            currentIndex = 0;
+            isLooping = loopSpawns;
+            if (isLooping)
+            {
+                // Only instantiate if needed
+                rng = new Random();
+            }
+        }
+
+        public void AddSkill(WeightedSkill skill)
+        {
+            Skills.Add(skill);
+            if (isLooping)
+            {
+                currentTotal += skill.Weight;
+            }
+        }
+
+        public void AddSkills(params WeightedSkill[] skills)
+        {
+            foreach (WeightedSkill skill in skills)
+            {
+                AddSkill(skill);
+            }
+        }
+
+        public Entity GetNextSkill()
+        {
+            if (Skills.Count == 0) return null;
+
+            Entity selectedSkill = null;
+            if (isLooping)
+            {
+                // Take the current index item
+                selectedSkill = Skills[currentIndex].Skill;
+                currentIndex = currentIndex == Skills.Count - 1 ? 0 : currentIndex + 1;
+            }

```

Comment:
To determine the next index, it might be nicer to use:
```cs
currentIndex = (currentIndex + 1) % Skills.Count;
```

Reply:
I thought verbose ver. was a bit more explicit in intent, but seeing it written out, this does look cleaner, so I will sub it in.